apps/advisor/pkg/app/checks/plugincheck/unsigned_step.go

```diff
@@ -0,0 +1,89 @@
+package plugincheck
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/grafana/grafana-app-sdk/logging"
+	advisor "github.com/grafana/grafana/apps/advisor/pkg/apis/advisor/v0alpha1"
+	"github.com/grafana/grafana/apps/advisor/pkg/app/checks"
+	"github.com/grafana/grafana/pkg/plugins"
+	"github.com/grafana/grafana/pkg/plugins/repo"
+)
+
+const (
+	UnsignedStepID = "unsigned"
+)
+
+type unsignedStep struct {
+	pluginIndex map[string]repo.PluginInfo
+}
+
+func (s *unsignedStep) Title() string {
+	return "Plugin signature check"
+}
+
+func (s *unsignedStep) Description() string {
+	return "Checks has a missing or invalid signature."
+}
+
+func (s *unsignedStep) Resolution() string {
+	return "Delete the plugin and install it from the Grafana plugin catalog."
```

Comment:
```suggestion
	return "For security, we recommend only installing plugins from the catalog. Review the plugin's status and verify your allowlist if appropriate."
```

I've struggled with this messaging I won't lie 😄